### DISCUSSION TITLE
Can I embed from a .qmd file in book mode?

### DISCUSSION THREAD

**Discussion by tpoisot** [quarto-dev/quarto-cli#6498] [Q&A]
### Description

I *do not* like notebooks. So when I saw in the documentation of the manuscript format that it was possible to embed a figure from a .qmd file, I was really excited!

I am trying to do this as part of a book project, and this line does not work:

```
{{< embed ../notebooks/sm-gradient-descent.qmd#fig-gradient-residuals >}}
```

The error I get is:

```
ERROR: Internal Error: Rendering an qmd notebook should only result in a single file. This attempt resulted in 0 file(s).
```

Is there a way to *not* use notebooks to embed results in a book format? This is doable for manuscripts, so it must be doable everywhere?

**Comment by tpoisot** [comment]
I *think* the issue is that quarto is trying to run the embedded file as an `ipynb` file, which is not the case?

**Reply by dragonstyle** [reply]
This does seem like a bug to me - we use the ipynb format as an intermediary format when embedding content from QMDs, but that shouldn't really affect things (we don't `run` the notebook, we run the QMD and render it to an ipynb which contains the outputs and code we need for embedding in a nice structure). I'm going to open an issue and take a look as soon as I can...

**Reply by dragonstyle** [reply]
(Issue is here: https://github.com/quarto-dev/quarto-cli/issues/6499)

**Reply by tpoisot** [reply]
Amazing, thanks for the prompt answer!
